laghos.cpp

```diff
@@ -585,38 +717,56 @@ int main(int argc, char *argv[])
 
          if (gfprint)
          {
-            ostringstream mesh_name, rho_name, v_name, e_name;
+            std::ostringstream mesh_name, rho_name, v_name, e_name;
             mesh_name << basename << "_" << ti
-                      << "_mesh." << setfill('0') << setw(6) << myid;
+                      << "_mesh." << std::setfill('0') << std::setw(6) << myid;
             rho_name  << basename << "_" << ti
-                      << "_rho." << setfill('0') << setw(6) << myid;
+                      << "_rho." << std::setfill('0') << std::setw(6) << myid;
             v_name << basename << "_" << ti
-                   << "_v." << setfill('0') << setw(6) << myid;
+                   << "_v." << std::setfill('0') << std::setw(6) << myid;
             e_name << basename << "_" << ti
-                   << "_e." << setfill('0') << setw(6) << myid;
+                   << "_e." << std::setfill('0') << std::setw(6) << myid;
 
-            ofstream mesh_ofs(mesh_name.str().c_str());
+            std::ofstream mesh_ofs(mesh_name.str().c_str());
             mesh_ofs.precision(8);
             pmesh->Print(mesh_ofs);
             mesh_ofs.close();
 
-            ofstream rho_ofs(rho_name.str().c_str());
+            std::ofstream rho_ofs(rho_name.str().c_str());
             rho_ofs.precision(8);
             rho_gf.Save(rho_ofs);
             rho_ofs.close();
 
-            ofstream v_ofs(v_name.str().c_str());
+            std::ofstream v_ofs(v_name.str().c_str());
             v_ofs.precision(8);
             v_gf.Save(v_ofs);
             v_ofs.close();
 
-            ofstream e_ofs(e_name.str().c_str());
+            std::ofstream e_ofs(e_name.str().c_str());
             e_ofs.precision(8);
             e_gf.Save(e_ofs);
             e_ofs.close();
          }
       }
+
+      // Problems checks
+      if (check)
+      {
+         double lnorm = e_gf * e_gf, norm;
+         MPI_Allreduce(&lnorm, &norm, 1, MPI_DOUBLE, MPI_SUM, pmesh->GetComm());
+         const double e_norm = sqrt(norm);
+         MFEM_VERIFY(rs_levels==0 && rp_levels==0, "check: rs, rp");
+         MFEM_VERIFY(order_v==2, "check: order_v");
+         MFEM_VERIFY(order_e==1, "check: order_e");
+         MFEM_VERIFY(ode_solver_type==4, "check: ode_solver_type");
+         MFEM_VERIFY(t_final == 0.6, "check: t_final");
+         MFEM_VERIFY(cfl==0.5, "check: cfl");
+         MFEM_VERIFY(strncmp(mesh_file, "default", 7) == 0, "check: mesh_file");
+         MFEM_VERIFY(dim==2 || dim==3, "check: dimension");
+         Checks(dim, ti, e_norm, checks);
```

Comment:
Are the `e_norm` numbers in checks independent of the device?

Reply:
Yes, I'm trying to have all the results independent of the PA/FA/Device options.

Reply:
Maybe I misunderstand, but I will not expect identicall results between PA and FA (e.g. both ob CPU); or between PA on CPU and PA on GPU...

Reply:
For the non-regression tests, that is useful though.